src/services/productService.js

```diff
@@ -14,3 +18,23 @@ export default function getProducts() {
       throw error.message;
     });
 }
+
+export function addProduct(productData) {
+  return fetch(PRODUCT_API, {
+    method: "POST",
+    headers: {
+      Authorization: `Token ${localStorage.getItem("token")}`,
+    },
+    body: productData,
+  }).then((res) => {
+    if (res.status === 201) return res.json();
+    throw res.json();
+  });
+}
+
+export function getProduct(id) {
+  return fetch(`${PRODUCT_API}${id}`).then((res) => {
+    if (res.status === 200) return res.json();
+    throw new DoesNotExistException("Product does not exist");
+  });
+}
```

Comment:
Handle errors here properly, to keep services consistent, either you receive an error message or you receive data.